src/receivers/ExpressReceiver.ts

```diff
@@ -327,25 +327,25 @@ export default class ExpressReceiver implements Receiver {
           return reject(new ReceiverInconsistentStateError(missingServerErrorDescription));
         }
 
-        resolve(this.server);
+        return resolve(this.server);
       });
     });
   }
 
   // TODO: the arguments should be defined as the arguments to close() (which happen to be none), but for sake of
   // generic types
   public stop(): Promise<void> {
+    if (this.server === undefined) {
+      return Promise.reject(new ReceiverInconsistentStateError('The receiver cannot be stopped because it was not started.'));
+    }
     return new Promise((resolve, reject) => {
-      if (this.server === undefined) {
-        return reject(new ReceiverInconsistentStateError('The receiver cannot be stopped because it was not started.'));
-      }
-      this.server.close((error) => {
+      this.server?.close((error) => {
```

Comment:
Hm. I assume `?` was introduced because it no longer connects the check done above with the logic that follows. Can that addition be avoided if we added an `else` with the alternative path inside?

Reply:
@misscoded so if I remove the `?`, when compiling the code (and, in particular, when trying to run the tests via `npm run mocha`), I get this:

```
➜ npm run mocha

> @slack/bolt@3.5.0 mocha /Users/fmaj/src/bolt-js
> TS_NODE_PROJECT=tsconfig.json nyc mocha --config .mocharc.json "src/**/*.spec.ts"


/Users/fmaj/src/bolt-js/node_modules/ts-node/src/index.ts:434
    return new TSError(diagnosticText, diagnosticCodes)
           ^
TSError: ⨯ Unable to compile TypeScript:
src/receivers/ExpressReceiver.ts:342:7 - error TS2532: Object is possibly 'undefined'.

342       this.server.close((error) => {
```

... which is silly, since we have a check for `undefined` right before, but I guess because of the intervening Promise function scope, the TS compiler can't assume that the two scopes are executed in the same context / immediately after one another? Not sure.

If anyone has suggestions how to avoid this, I'd love to learn!